set.mm

```diff
@@ -215251,6 +215253,65 @@ is an important property of monoids (see ~ mndid ), and therefore also for
       IUGUJUKULUMUNAHUOTUPUSVEACVGUQTUR $.
   $}
 
+  ${
+    $d n u v w x y z B $.  $d n u v w x y z O $.  $d n u v w x y z ph $.
+    $d u v w y z N $.  $d n u v w x y z .+ $.  $d u v w y z X $.
+    $d u v w y ps $.
+    grprinvlem.c $e |- ( ( ph /\ x e. B /\ y e. B ) -> ( x .+ y ) e. B ) $.
+    grprinvlem.o $e |- ( ph -> O e. B ) $.
+    grprinvlem.i $e |- ( ( ph /\ x e. B ) -> ( O .+ x ) = x ) $.
+    grprinvlem.a $e |- ( ( ph /\ ( x e. B /\ y e. B /\ z e. B ) )
+          -> ( ( x .+ y ) .+ z ) = ( x .+ ( y .+ z ) ) ) $.
+    grprinvlem.n $e |- ( ( ph /\ x e. B ) -> E. y e. B ( y .+ x ) = O ) $.
+    ${
+      grprinvlem.x $e |- ( ( ph /\ ps ) -> X e. B ) $.
+      grprinvlem.e $e |- ( ( ph /\ ps ) -> ( X .+ X ) = X ) $.
+      $( Lemma for ~ grprinvd .  (Contributed by NM, 9-Aug-2013.) $)
+      grprinvlem $p |- ( ( ph /\ ps ) -> X = O ) $=
+        ( cv co wceq wcel vu vv vw wa wrex wral ralrimiva oveq2 rexbidv cbvralv
+        eqeq1d sylib syl2an2r oveq2d adantr simprr oveq1d w3a caovassg ad4ant14
+        rspccva simprl caovassd id eqeq12d rspcdva 3eqtr3d rexlimddv ) ABUDZDQZ
+        IGRZHSZIHSDFAVJEQZGRZHSZDFUEZEFUFZBIFTZVLDFUEZAVJCQZGRZHSZDFUEZCFUFVQAW
+        CCFNUGWCVPCEFVTVMSZWBVODFWDWAVNHVTVMVJGUHUKUIUJULOVPVSEIFVMISZVOVLDFWEV
+        NVKHVMIVJGUHUKUIVAUMVIVJFTZVLUDZUDZVJIIGRZGRZVKIHVIWJVKSWGVIWIIVJGPUNUO
+        WHVKIGRHIGRZWJIWHVKHIGVIWFVLUPZUQWHUAUBUCVJIIFGAUAQZFTUBQZFTUCQZFTURWMW
+        NGRWOGRWMWNWOGRGRSBWGACDEWMWNWOFGMUSUTVIWFVLVBVIVRWGOUOZWPVCVIWKISZWGVI
+        HVJGRZVJSZWQDFIVJISZWRWKVJIVJIHGUHWTVDVEAWSDFUFZBAHVTGRZVTSZCFUFXAAXCCF
+        LUGXCWSCDFVTVJSZXBWRVTVJVTVJHGUHXDVDVEUJULUOOVFUOVGWLVGVH $.
+    $}
+
+    ${
```

Comment:
Could you close the scope of the lemma, and open another scope for the theorem, with explicitly stated $e hypotheses ? Have large scopes makes it difficult to know which hypotheses are used/necessary where.

Reply:
This is not created by me, I just moved this theorems. And I think it is a usual method to nest theorems in this way. Unless there is no decission to change this in general, I would leave it here as it is.